pkg/processor/reportprocessor.go

```diff
@@ -0,0 +1,231 @@
+	getResp := restClient.Get().
+	    Resource("policyreports").
+		Namespace(cluster.Namespace).
+		Name(ruleName).
+		Do(context.TODO())
+
+	respBytes, _ := getResp.Raw()
+	var prResponse types.PolicyReportGetResponse
+	unmarshalError := json.Unmarshal(respBytes, &prResponse)
+	if unmarshalError != nil {
+		glog.Infof(
+			"Error unmarshalling PolicyReport: %v",
+			unmarshalError,
+		)
+	}
+
+	if (prResponse.Meta.Name == "") {
+		// If the PolicyReport doesn't exist Create it
+		// TODO Need to use report.details to fill in the template values
+		// Example: policyReport.Summary may have template strings that need to be replaced by the report.details information
+		policyreport := &v1alpha1.PolicyReport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ruleName,
+				Namespace: cluster.Namespace,
+			},
+			Results: []*v1alpha1.PolicyReportResult{{
+				Policy:   report.Key,
+				Message:  policyReport.Description,
+				// We will use Scored to represent whether the violation has been resolved
+				// On creation it is false, when the violation is cleared it is set to true
+				Scored:   false,
+				Category: strings.Join(policyReport.Tags, ","),
+				Status:   "error",
+				Data: map[string]string{
+					"created_at": policyReport.Publish_date,
+					// TODO total_risk is no longer available, need to sync with CCX team to determine best route here
+					"total_risk": strconv.Itoa(policyReport.Likelihood),
+					"reason":     policyReport.Reason,
+					"resolution": policyReport.Resolution,
+				},
+			}},
+		}
+
+		postResp := restClient.Post().
+			Resource("policyreports").
+			Namespace(cluster.Namespace).
+			Body(policyreport).
+			Do(context.TODO())
+
+		if postResp.Error() != nil {
+			glog.Infof(
+				"Error creating PolicyReport %s for cluster %s (%s): %v",
+				report.Component,
+				cluster.Namespace,
+				cluster.ClusterID,
+				postResp.Error(),
+			)
+		} else {
+			glog.Infof(
+				"Successfully created PolicyReport %s for cluster %s (%s)",
+				report.Component,
+				cluster.Namespace,
+				cluster.ClusterID,
+			)
+		}
+	} else if (prResponse.Meta.Name != "" && prResponse.Results[0].Status == "skip") {
+		glog.Infof("PolicyReport %s has been reintroduced, updating status to error", prResponse.Meta.Name)
+		payload := []patchStringValue{{
+			Op:    "replace",
+			Path:  "/results/0/status",
+			Value: "error",
+		}}
+		payloadBytes, _ := json.Marshal(payload)
+
+		resp := restClient.Patch(k8sTypes.JSONPatchType).
+			Resource("policyreports").
+			Namespace(cluster.Namespace).
+			Name(ruleName).
+			Body(payloadBytes).
+			Do(context.TODO())
+
+		if resp.Error() != nil {
+			glog.Infof(
+				"Error updating PolicyReport %s status to error for cluster %s: %v",
+				report.Component,
+				cluster.Namespace,
+				resp.Error(),
+			)
+		} else {
+			glog.Infof(
+				"Successfully updated PolicyReport %s status to error for cluster %s",
+				report.Component,
+				cluster.Namespace,
+			)
+		}
+	}
```

Comment:
Need to handle the case where the status is error and we needed to update other field.